src/EditorFeatures/Core.Wpf/Adornments/AdornmentManager.cs

```diff
@@ -179,19 +180,14 @@ private void OnTagsChanged(object sender, TagsChangedEventArgs e)
                 if (needToScheduleUpdate)
                 {
                     // schedule an update
-                    var asyncToken = _asyncListener.BeginAsyncOperation(GetType() + ".OnTagsChanged.2");
-                    _textView.VisualElement.Dispatcher.BeginInvoke(
-                        new System.Action(() =>
+                    _threadingContext.JoinableTaskFactory.WithPriority(_textView.VisualElement.Dispatcher, DispatcherPriority.Render).RunAsync(async () =>
                     {
-                        try
+                        using (_asyncListener.BeginAsyncOperation(GetType() + ".OnTagsChanged.2"))
                         {
+                            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.